### DISCUSSION TITLE
Weak Ciphers Enabled

### DISCUSSION THREAD

**Discussion by YaroslavVoloshchuk** [nginx-proxy/nginx-proxy#2326] [Q&A]
Hello! 

Scan report shows:

List of Supported Weak Ciphers
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256 (0xC027)
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384 (0xC028)

I want to avoid it. How to fix?

Image: jwilder/nginx-proxy:alpine

In a docker container /etc/nginx/conf.d file default.conf include by default:

    ssl_protocols TLSv1.2 TLSv1.3;
    ssl_ciphers 'ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-CHACHA20-POLY1305';
    ssl_prefer_server_ciphers off;

I 've tried to add to global.conf changes:
ssl_ciphers 'ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384';

But get an error:
nginx.1     | 2023/10/31 11:19:10 [error] 25#25: *2 open() "/usr/share/nginx/html/health" failed (2: No such file or directory), client: 172.18.24.6, server: localhost, request: "GET /health HTTP/1.1", host: "172.18.24.38"

or

nginx.1     | nginx: [emerg] "server_tokens" directive is duplicate in /etc/nginx/conf.d/global.conf:1

Please help me. Thanks

**Comment by buchdag** [comment]
Hi. You'll find instruction on how to change the cipher configuration [here](https://github.com/nginx-proxy/nginx-proxy#how-ssl-support-works).
